controller/replica_controller.go

```diff
@@ -179,6 +199,54 @@ func (rc *ReplicaController) handleErr(err error, key interface{}) {
 	rc.queue.Forget(key)
 }
 
+// From replica to check Node.Spec.EvictionRequested of the node
+// this replica first, then check Node.Spec.Disks.EvictionRequested
+func (rc *ReplicaController) isEvictionRequested(replica *longhorn.Replica) bool {
+	metaObj, err := meta.Accessor(replica)
+	if err != nil {
+		logrus.Errorf("BUG: %v cannot be convert to metav1.Object: %v", replica, err)
+		return false
+	}
+	labels := metaObj.GetLabels()
+	node, err := rc.ds.GetNode(labels["longhornnode"])
```

Comment:
I am not sure why you prefer to use the label to check and find the corresponding node? Can we directly use `replica.Spec.NodeID` instead?

Reply:
Ok, I see. I just found it in the instanceSpec. I will update this part.